packages/toolkit/src/query/react/buildHooks.ts

```diff
@@ -701,12 +701,21 @@ export function buildHooks<Definitions extends EndpointDefinitions>({
       const promiseRef = useRef<QueryActionCreatorResult<any>>()
 
       let { queryCacheKey, requestId } = promiseRef.current || {}
-      const currentRenderHasSubscription = useSelector(
-        (state: RootState<Definitions, string, string>) =>
-          !!queryCacheKey &&
-          !!requestId &&
-          !state[api.reducerPath].subscriptions[queryCacheKey]?.[requestId]
-      )
+
+      // HACK Because the latest state is in the middleware, we actually
```

Comment:
Because of this, it is not really problematic if the user forgets to add the middleware to the store.
Before, there was just no cleanup happening. Now, we get unexpected behaviour. Maybe we should change that "you forgot to add the middleware" warning to throwing an error at this points?

Reply:
Addressed in #2792